checksec.cpp

```diff
@@ -161,6 +162,65 @@ Checksec::operator json() const {
     };
 }
 
+const std::string Checksec::detailedReport() const {
+    json j = toJson();
+    std::string s = "";
```

Comment:
Since you mentioned not being experienced with C++: the idiomatic way to do this kind of function is with a `std::stringstream`, which you can then feed chunks with the stream-style `<<` operator. You can then pull the ultimate `std::string` out of the stream with `std::stringstream::str()`.